### PR TITLE
Ship assert_fpm script

### PR DESCRIPTION
**Pull request by giorgiosironi**
Higher level than `curl_fpm`, to avoid reinventing the wheel in each child image.

Can't yet run the base smoke test during the build, because it will need some orchestration with docker-compose to get the container up, run an additional script and then taking it down.

### CODE REVIEW

File: utils/assert_fpm [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+output=$(SCRIPT_FILENAME="${PROJECT_FOLDER}/${PHP_ENTRYPOINT}${path}" REQUEST_URI="${path}" REQUEST_METHOD=GET cgi-fcgi -bind -connect 127.0.0.1:9000)
+fail() {
+    echo "---Full output follows---"
+    echo "${output}"
+    exit 2
+}
+
+status=$(echo "${output}" | grep -o "Status: [0-9]\+ " | sed -e "s/[^0-9]\+//g")
+if [ -z "$status" ]; then
+    status=200
```
Comment (thewilkybarkid) [inline utils/assert_fpm]: Removes non numbers from the status, and if it's not found treat it as 200?

File: utils/assert_fpm [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+if [ -z "${PHP_ENTRYPOINT}" ]; then
+    echo "PHP_ENTRYPOINT environment variable must be set"
+    echo "Example: PHP_ENTRYPOINT=web/app.php"
+    exit 1
+fi
+
+path="$1"
+text="$2"
+
+output=$(SCRIPT_FILENAME="${PROJECT_FOLDER}/${PHP_ENTRYPOINT}${path}" REQUEST_URI="${path}" REQUEST_METHOD=GET cgi-fcgi -bind -connect 127.0.0.1:9000)
```
Comment (thewilkybarkid) [inline utils/assert_fpm]: Just tried the script and `${path}` shouldn't be in `SCRIPT_FILENAME`?

File: hello_world.php [hunk 3]
```diff
@@ -0,0 +1,2 @@
+<?php
+echo "Hello world";
```
Comment (thewilkybarkid) [inline hello_world.php]: Probably unnecessary, but we could replicate the standard `/ping` path.

```php
<?php

http_response_code(200);
header('Cache-Control: must-revalidate, no-cache, no-store, private');
header('Content-Type: text/plain; charset=UTF-8');

echo 'pong';
```

File: utils/assert_fpm (hunk 1 above)
Comment (giorgiosironi) [reply utils/assert_fpm]: It seems that 200 is not printed as a status:
```
$ docker exec -it d9ae355 /bin/bash -c 'SCRIPT_FILENAME=/var/www/html/hello_world.php REQUEST_URI=/ REQUEST_METHOD=GET cgi-fcgi -bind -connect 127.0.0.1:9000'
X-Powered-By: PHP/7.0.27
Content-type: text/html; charset=UTF-8

Hello world
$ docker exec -it d9ae355 /bin/bash -c 'SCRIPT_FILENAME=/var/www/html/hello_wor REQUEST_URI=/ REQUEST_METHOD=GET cgi-fcgi -bind -connect 127.0.0.1:9000'
Primary script unknown
Status: 404 Not Found
X-Powered-By: PHP/7.0.27
Content-type: text/html; charset=UTF-8

File not found.
```
but I can't find a specification, purely experimenting.

File: utils/assert_fpm (hunk 2 above)
Comment (giorgiosironi) [reply utils/assert_fpm]: Comparing to https://github.com/elifesciences/journal-formula/blob/master/salt/journal/config/etc-nginx-sites-available-journal.conf#L86, yes, it should not be there

File: utils/assert_fpm (hunk 1 above)
Comment (giorgiosironi) [reply utils/assert_fpm]: Like, http://www.mit.edu/~yandros/doc/specs/fcgi-spec.html doesn't mention `SCRIPT_FILENAME` at all

**Comment by giorgiosironi** [comment]
Chosen a `DEBUG` environment variable (similar to `$config['debug']`) which will only be created by `ci` images.

File: utils/assert_fpm [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    echo "${output}"
+    exit 2
+}
+
+status=$(echo "${output}" | grep -o "Status: [0-9]\+ " | sed -e "s/[^0-9]\+//g")
+if [ -z "$status" ]; then
+    status=200
+fi
+echo "Status: $status"
+if [ $status -gt 399 ]; then
```
Comment (thewilkybarkid) [inline utils/assert_fpm]: Maybe not 200? Anything else would be wrong...

Comment (giorgiosironi) [reply utils/assert_fpm]: We had a case of `202` in `personalised-covers` once (which then was changed to `200` later). If I wanted to test a redirect, I would not use `assert_fpm` or maybe I would provide some other parameter. Changing it to `-gt 299`.

**Pull request merged**